packages/legend-graph/src/graphManager/protocol/pure/v1/V1_PureGraphManager.ts

```diff
@@ -769,6 +775,59 @@ export class V1_PureGraphManager extends AbstractPureGraphManager {
         options,
       );
 
+      // Resolve function's functionName and parameter/returnValue's type/multiplicity
+      await Promise.all<void>(
+        buildInputs.map((input) =>
+          promisify(() => {
+            input.data.nativeElements
+              .filter(
+                (element) => element instanceof V1_ConcreteFunctionDefinition,
+              )
+              .forEach((element) => {
+                try {
+                  const context = this.getBuilderContext(
+                    graph,
+                    input.model,
+                    element,
+                    options,
+                  );
+                  const func = context.currentSubGraph.getOwnFunction(
+                    V1_buildFullPath(
+                      element.package,
+                      V1_buildFunctionSignature(
+                        element as V1_ConcreteFunctionDefinition,
+                      ),
+                    ),
+                  );
+                  func.returnType = context.resolveType(
```

Comment:
the point of the light graph is to only have to index the graph. is there any way we can display the name without having to essentially compile the function?

Reply:
I am afraid no.. because when we displaying the name based on the metaModel. To build correct function pretty name, we must need information about returnType + multiplicity, parameter type + multiplicity, and functionName.